From: Simon McVittie <smcv@collabora.com>
Date: Tue, 13 Feb 2024 11:33:29 +0000
Subject: girepository test: Don't assume doubles are naturally-aligned

On i386 Linux, the minimal ABI alignment of a double is only 4, and
therefore the alignment of GDoubleIEEE754 is also 4.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3923
Applied-upstream: 2.79.3, commit:2b2b6de70bf01dfe70b69dc879b2c929ffb79b8a
---
 girepository/tests/union-info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/girepository/tests/union-info.c b/girepository/tests/union-info.c
index 7eca2b6..6dfa78b 100644
--- a/girepository/tests/union-info.c
+++ b/girepository/tests/union-info.c
@@ -52,7 +52,7 @@ test_basic (RepositoryFixture *fx,
   g_assert_null (gi_union_info_get_discriminator (double_info, 0));
 
   g_assert_cmpuint (gi_union_info_get_size (double_info), ==, 8);
-  g_assert_cmpuint (gi_union_info_get_alignment (double_info), ==, 8);
+  g_assert_cmpuint (gi_union_info_get_alignment (double_info), ==, G_ALIGNOF (GDoubleIEEE754));
 
   g_assert_null (gi_union_info_get_copy_function_name (double_info));
   g_assert_null (gi_union_info_get_free_function_name (double_info));
